File: app/src/main/java/wseemann/media/romote/receiver/CommandReceiver.java
package wseemann.media.romote.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.wseemann.ecp.api.ResponseCallback;
import com.wseemann.ecp.core.KeyPressKeyValues;
import com.wseemann.ecp.request.KeyPressRequest;
import java.io.UnsupportedEncodingException;
import javax.inject.Inject;
import dagger.hilt.android.AndroidEntryPoint;
import timber.log.Timber;
import wseemann.media.romote.utils.CommandHelper;

@AndroidEntryPoint
public class CommandReceiver extends BroadcastReceiver {

    @Inject
    protected CommandHelper commandHelper;

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent != null) {
            String url = commandHelper.getDeviceURL();
            KeyPressKeyValues keypressKeyValues = (KeyPressKeyValues) intent.getSerializableExtra("keypress");

            if (keypressKeyValues == null) {
                return;
            }

            try {
                KeyPressRequest keypressRequest = new KeyPressRequest(url, keypressKeyValues.getValue());
                keypressRequest.sendAsync(new ResponseCallback<>() {
                    @Override
                    public void onSuccess(@Nullable Void unused) {

                    }

                    @Override
                    public void onError(@NonNull Exception e) {

                    }
                });
            } catch (UnsupportedEncodingException ex) {
                Timber.e(ex, "Failed to execute command");
            }
        }
    }
}

File: app/src/main/java/wseemann/media/romote/receiver/CommandReceiver.kt
package wseemann.media.romote.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.wseemann.ecp.core.KeyPressKeyValues
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.CoroutineDispatcher
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.launch
import timber.log.Timber
import wseemann.media.romote.device.DeviceManager
import wseemann.media.romote.di.IoDispatcher
import javax.inject.Inject

@AndroidEntryPoint
class CommandReceiver : BroadcastReceiver() {

    @Inject
    @IoDispatcher
    lateinit var ioDispatcher: CoroutineDispatcher

    @Inject
    lateinit var deviceManager: DeviceManager

    override fun onReceive(context: Context, intent: Intent?) {
        if (intent == null) return

        val keypressKeyValues = intent.getSerializableExtra("keypress") as? KeyPressKeyValues
        if (keypressKeyValues == null) {
            Timber.w("Received intent without a valid 'keypress' extra, ignoring")
            return
        }

        CoroutineScope(ioDispatcher).launch {
            deviceManager.getConnectedDevice()?.performKeyPress(keypressKeyValues)
        }
    }
}
